File: CaseStudy/stock/src/main/java/com/stores/stock/model/Stock.java
package com.stores.stock.model;

import javax.persistence.*;

@Entity
@Table(name="Stock")
public class Stock {
    @Id
    private int storeId;
    @Id
    private int productId;
    private int stock;

    public int getStoreId() {
        return storeId;
    }

    public void setStoreId(int storeId) {
        this.storeId = storeId;
    }

    public int getProductId() {
        return productId;
    }

    public void setProductId(int productId) {
        this.productId = productId;
    }

    public int getStock() {
        return stock;
    }

    public void setStock(int stock) {
        this.stock = stock;
    }

    public Stock(int storeId, int productId, int stock) {
        this.storeId = storeId;
        this.productId = productId;
        this.stock = stock;
    }

    public Stock() {
    }

    @Override
    public String toString() {
        return "Stock{" +
                "storeId=" + storeId +
                ", productId=" + productId +
                ", stock=" + stock +
                '}';
    }
}

File: CaseStudyKotlin/demo/src/main/kotlin/com/store/demo/model/Stock.kt
package com.store.demo.model

import java.io.Serializable
import java.util.*
import javax.persistence.*
@Entity
@Table(name = "Stock")
@IdClass(Stock::class)
class Stock : Serializable {
    @Id
    @PrimaryKeyJoinColumn
    var storeId = 0

    @Id
    @PrimaryKeyJoinColumn
    var productId = 0
    var stock = 0

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o !is Stock) return false
        val stock1 = o
        return storeId == stock1.storeId && productId == stock1.productId && stock == stock1.stock
    }

    override fun hashCode(): Int {
        return Objects.hash(storeId, productId, stock)
    }

    constructor(storeId: Int, productId: Int, stock: Int) {
        this.storeId = storeId
        this.productId = productId
        this.stock = stock
    }

    constructor() {}

    override fun toString(): String {
        return "Stock{" +
                "storeId=" + storeId +
                ", productId=" + productId +
                ", stock=" + stock +
                '}'
    }
}